src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrayToStyles = sourceString
+    .trim()
+    .split(';')
+    .filter(styleProp => styleProp.includes(':'))
+    .reduce((stylesList, stylesParameters) => {
+      const [property, value] = stylesParameters.split(':');
+
+      if (property !== undefined && value !== undefined) {
+        return {
+          ...stylesList,
+          [property.trim()]: value.trim(),
+        };
```

Comment:
It would be better to not create a new object and iterate through the styles list when using spread every time.

```suggestion
        stylesList[property.trim()] = value.trim();
```